addons/mail/static/src/models/thread/thread.js

```diff
@@ -1077,15 +1082,15 @@ function factory(dependencies) {
             // serverMessageUnreadCounter
             if (this.lastSeenByCurrentPartnerMessageId < firstMessage.id) {
                 const fetchedNotSeenMessages = this.orderedMessages.filter(message =>
-                    message.id > this.serverLastMessageId
+                    !message.isTransient && message.id > this.serverLastMessageId
                 );
                 return this.serverMessageUnreadCounter + fetchedNotSeenMessages.length;
             }
             // lastSeenByCurrentPartnerMessageId is a known message,
             // then we can forget serverMessageUnreadCounter
             const maxId = Math.max(this.serverLastMessageId, this.lastSeenByCurrentPartnerMessageId);
             return this.orderedMessages.reduce(
-                (acc, message) => acc + (message.id > maxId ? 1 : 0),
+                (acc, message) => acc + ((!message.isTransient && message.id > maxId) ? 1 : 0),
```

Comment:
This prevents flickering (1) in chat window

Reply:
Maybe we want a computed field that is like `orderedMessages` but without `isTransient` messages on it? And then use that in all the places where that makes sense?

Reply:
Maybe but how would you name that field lol ? "realOrderedMessages" ?

Reply:
I would keep the "real" one with the existing name probably, and introduce a new one such as `orderedMessagesIncludingTransient`?

Honestly I just dislike the concept of fake records, and we are really starting to see why.